docs/tendermint-core/consensus/proposer-based-timestamps.md

```diff
@@ -0,0 +1,89 @@
+
+### `Precision`
+
+The `Precision` parameter configures the acceptable upper-bound of clock drift among
+all of the nodes on a Tendermint network. Any two nodes on a Tendermint network
+are expected to have clocks that differ by at most `Precision` seconds? Ms? at
+any given instant.
+
+### `MessageDelay`
+
+The `MessageDelay` parameter configures the acceptable upper-bound for transmitting
+a `Proposal` message from the proposer to _all_ of the validators on the network.
+
+Networks that want very tight bounds on their timestamps should set a smaller value
+for `MessageDelay`. Highly distributed networks or networks with generally slow
+rates of transmission should set a larger value for `MessageDelay`.
+
+## Algorithm Concepts
+
+### Block timestamps
+
+Each block produced by the Tendermint consensus engine contains a timestamp.
+The timestamp produced in each block is a meaningful representation of time that is
+useful for the protocols and applications built on top of Tendermint.
+
+The following protocols and application features require a reliable source of time:
+
+* Tendermint Light Clients [rely on correspondence between their known time](https://github.com/tendermint/tendermint/blob/master/spec/light-client/verification/README.md#definitions-1) and the block time for block verification.
+* Tendermint Evidence validity is determined [either in terms of heights or in terms of time](https://github.com/tendermint/tendermint/blob/master/spec/consensus/evidence.md#verification).
+* Unbonding of staked assets in the Cosmos Hub [occurs after a period of 21 days](https://github.com/cosmos/governance/blob/master/params-change/Staking.md#unbondingtime).
+* IBC packets can use either a [timestamp or a height to timeout packet delivery](https://docs.cosmos.network/v0.44/ibc/overview.html#acknowledgements)
+
+### Proposer Selects a Block Timestamp
+
+When the proposer node creates a new block proposal, the node reads the time from 
+its local clock and uses this reading as the timestamp for the proposed block.
+
+### Timely-ness
+
+When each validator on a Tendermint network receives a proposed block, it performs
+a series of checks to ensure that the block can be considered valid as a candidate
+to be the next block in the chain.
+
+The Proposer-Based Timestamps algorithm performs a validity check on the timestamp of proposed blocks.
+When a validator receives a proposal it ensures that the timestamp in the proposal
+is within a bound of the validator's local clock. Specifically, the algorithm
+checks that the timestamp is no more than `Precision` greater than the node's local clock
+and no less than `Precision` + `MessageDelay` behind than the node's local clock.
+This creates range of acceptable timestamps around the node's local time. If
+the timestamp is within this range, the Proposer-Based Timestamps algorithm considers
+the block `timely`. If a block is not `timely`, the node will issue a `nil` `prevote`
```

Comment:
```suggestion
the block **timely**. If a block is not **timely**, the node will issue a `nil` prevote
```

I think it's worthwhile to use different typography for `code` vs. **terminology** (or _terminology_). I'd also suggest it's not necessary to set off every use of a term, unless it's related to the current topic.